src/components/ScreenWrapper/index.js

```diff
@@ -1,155 +1,156 @@
+            const shouldDismissKeyboard = shouldDismissKeyboardBeforeClose && isKeyboardShown && Browser.isMobile();
 
-        this.state = {
-            didScreenTransitionEnd: false,
-        };
-    }
+            return isHorizontalSwipe && shouldDismissKeyboard;
+        },
+        onPanResponderGrant: Keyboard.dismiss,
+    });
 
-    componentDidMount() {
-        this.unsubscribeTransitionEnd = this.props.navigation.addListener('transitionEnd', (event) => {
+    useEffect(() => {
+        const unsubscribeTransitionEnd = navigation.addListener('transitionEnd', (event) => {
             // Prevent firing the prop callback when user is exiting the page.
             if (lodashGet(event, 'data.closing')) {
                 return;
             }
-            this.setState({didScreenTransitionEnd: true});
-            this.props.onEntryTransitionEnd();
+
+            setDidScreenTransitionEnd(true);
+            onEntryTransitionEnd();
         });
 
         // We need to have this prop to remove keyboard before going away from the screen, to avoid previous screen look weird for a brief moment,
         // also we need to have generic control in future - to prevent closing keyboard for some rare cases in which beforeRemove has limitations
         // described here https://reactnavigation.org/docs/preventing-going-back/#limitations
-        if (this.props.shouldDismissKeyboardBeforeClose) {
-            this.beforeRemoveSubscription = this.props.navigation.addListener('beforeRemove', () => {
-                if (!this.props.isKeyboardShown) {
-                    return;
-                }
-                Keyboard.dismiss();
-            });
-        }
-    }
+        const beforeRemoveSubscription = shouldDismissKeyboardBeforeClose
+            ? navigation.addListener('beforeRemove', () => {
+                  if (!isKeyboardShown) {
+                      return;
+                  }
+                  Keyboard.dismiss();
+              })
+            : undefined;
 
-    /**
-     * We explicitly want to ignore if props.modal changes, and only want to rerender if
-     * any of the other props **used for the rendering output** is changed.
-     * @param {Object} nextProps
-     * @param {Object} nextState
-     * @returns {boolean}
-     */
-    shouldComponentUpdate(nextProps, nextState) {
-        return !_.isEqual(this.state, nextState) || !_.isEqual(_.omit(this.props, 'modal'), _.omit(nextProps, 'modal'));
-    }
+        return () => {
+            unsubscribeTransitionEnd();
 
-    componentWillUnmount() {
-        if (this.unsubscribeTransitionEnd) {
-            this.unsubscribeTransitionEnd();
-        }
-        if (this.beforeRemoveSubscription) {
-            this.beforeRemoveSubscription();
-        }
-    }
-
-    render() {
-        const maxHeight = this.props.shouldEnableMaxHeight ? this.props.windowHeight : undefined;
+            if (typeof beforeRemoveSubscription === 'function') {
```

Comment:
```suggestion
            if (beforeRemoveSubscription) {
```

I think you can just check if `beforeRemoveSubscription` is truthy